packages/react/src/date-pickers/DatePicker/DatePickerToggle.js

```diff
@@ -63,11 +69,10 @@ const DatePickerToggle = forwardRef((
     'aria-haspopup': 'menu',
     disabled,
     id: datePickerToggleId,
-    onClick: handleClick,
-    onKeyDown: handleKeyDown,
+    onClick: callEventHandlers(onClickProp, onClick),
+    onKeyDown: callEventHandlers(onKeyDownProp, onKeyDown),
     ref: combinedRef,
-    role: 'button',
-    tabIndex: 0,
+    role: 'group',
```

Comment:
Note: Change the toggle role from `button`  to `group` to follow the MUI implementation

<img width="729" height="196" alt="image" src="https://github.com/user-attachments/assets/7d732e72-c4de-47aa-bac1-7b371cba6215" />